File: Notification-Service/src/test/java/byrd/riley/notificationservice/test/controller/UnitTest.java
package byrd.riley.notificationservice.test.controller;

import static org.junit.jupiter.api.Assertions.assertThrows;

import byrd.riley.notificationservice.model.SMSConfigurable;
import byrd.riley.notificationservice.model.Texter;
import byrd.riley.notificationservice.test.model.TestSMSConfiguration;
import org.junit.jupiter.api.Test;

import com.twilio.exception.ApiException;

import byrd.riley.notificationservice.model.TwilioTexter;

class UnitTest {

	private SMSConfigurable config = new TestSMSConfiguration();
	private Texter texter = new TwilioTexter(config);

	@Test
	void passingSendText() {
		texter.sendText(TestSMSConfiguration.VALID_RECEIVER_NUMBER, "Test message.");
	}

	@Test
	void failingSendText() {
		assertThrows(ApiException.class, () -> texter.sendText(TestSMSConfiguration.INVALID_RECEIVER_NUMBER, "Test message."));
	}
}

File: Notification-Service-Kotlin/src/test/kotlin/byrd/riley/notificationservicekotlin/test/controller/UnitTest.kt
package byrd.riley.notificationservicekotlin.test.controller

import byrd.riley.notificationservicekotlin.model.SMSConfigurable
import byrd.riley.notificationservicekotlin.test.model.TestSMSConfiguration
import byrd.riley.notificationservicekotlin.model.Texter
import byrd.riley.notificationservicekotlin.model.TwilioTexter
import com.twilio.exception.ApiException
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test

class UnitTest {

    private val config: SMSConfigurable = TestSMSConfiguration()
    private val texter: Texter = TwilioTexter(config)

    @Test
    fun passingSendText() {
        texter.sendText(TestSMSConfiguration.VALID_RECEIVER_NUMBER, "Test message.")
    }

    @Test
    fun failingSendText() {
        assertThrows(ApiException::class.java) { texter.sendText(TestSMSConfiguration.INVALID_RECEIVER_NUMBER, "Test message.") }
    }
}